packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts

```diff
@@ -587,21 +609,17 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
             const tombstoneContainer = await loadContainer(summaryVersion);
 
             // Datastore should be tombstoned, requesting should error
-            await assert.rejects(async () => requestFluidObject<ITestDataObject>(tombstoneContainer, unreferencedId),
-                (error) => {
-                    const correctErrorType = error.code === 404;
-                    const correctErrorMessage = error.message === `Datastore removed by gc: ${unreferencedId}`;
-                    return correctErrorType && correctErrorMessage;
-                },
-                `Should not be able to retrieve a tombstoned datastore.`,
-            );
+            const tombstoneResponse = await containerRuntime_resolveHandle(tombstoneContainer, { url: unreferencedId });
+            assert.equal(tombstoneResponse.status, 404, "Expected 404 tombstone response")
+            assert.equal(tombstoneResponse.headers?.[TombstoneResponseHeaderKey], true, "Expected tombstone response with flag")
 
-            // Normally, there should be a SweepReady_Revived error when referencing a SweepReady data store. However,
-            // since the summarizer client is the one that is created and adding the handle, the handle is not decoded
-            // and it will not result in GC detecting the reference. Also, a "gcUnknownOutboundReferences" error will
-            // be logged for the same reason.
-            const mainDataObject = await requestFluidObject<ITestDataObject>(summarizingContainer, "default");
-            mainDataObject._root.set("store", dataObject.handle);
+            const requestAllowingTombstone: IRequest = {
+                url: unreferencedId,
+                headers: { [AllowTombstoneRequestHeaderKey]: true },
+            };
+            const tombstonedObject = await requestFluidObject<ITestDataObject>(tombstoneContainer, requestAllowingTombstone);
```

Comment:
Why use requestFluidObject here and containerRuntime_resolveHandle above?

Reply:
I think anywhere that was getting the handle off the data object I used RequestFluidObject rather than writing that Provider code myself. It's a little arbitrary, definitely open to input.